testpyisomd5sum.py

```diff
@@ -12,18 +15,34 @@ def pass_fail(rc, pass_value, pass_all):
     else:
         return ("FAIL", False)
 
+try:
+    iso_size = int(sys.argv[1])
+except (IndexError, ValueError):
+    iso_size = 0
 
-# create iso file
 try:
     # Python 3
     catch_error = FileNotFoundError
 except NameError:
     # Python 2
     catch_error = OSError
-try:
-    subprocess.check_call(["mkisofs", "-quiet", "-o", "testiso.iso", "."])
-except catch_error:
-    subprocess.check_call(["genisoimage", "-quiet", "-o", "testiso.iso", "."])
+
+# create iso file using a clean directory
+with tempfile.TemporaryDirectory(prefix="isomd5test-") as tmpdir:
```

Comment:
`TemporaryDirectory` is Python 3.2+.  Personally I say kill 2.7 with fire, but I'll leave that decision to you. :)

(If you do decide to explicitly announce dropping 2.7 compatibility, I can prepare a PR which gets rid of the various 2/3 hacks.  I need to make a PR about the mkisofs/genisoimage code in the test anyway, to add xorriso support.)

Reply:
Yeah, it's time to drop py2 support. In Fedora we no longer build it for py2 so those bits are really just leftovers that haven't been cleaned up.